datafusion/physical-expr/src/expressions/try_cast.rs

```diff
@@ -82,6 +82,19 @@ impl PhysicalExpr for TryCastExpr {
                 &array,
                 &self.cast_type,
             )?)),
+            ColumnarValue::Scalar(scalar)
```

Comment:
This is the actual code change -- the rest of the PR is tests

Reply:
Won't this prevent this from using the scalar comparison kernels, effectively reversing #2808

Reply:
🤔 that is a good point -- perhaps I'll have a go at a proper fix for https://github.com/apache/arrow-datafusion/issues/2874